src/sdk/pynni/nni/compression/torch/pruning/structured_pruning.py

```diff
@@ -78,17 +113,142 @@ def calc_mask(self, sparsity, wrapper, wrapper_idx=None):
         num_prune = int(num_total * sparsity)
         if self.preserve_round > 1:
             num_preserve = num_total - num_prune
-            num_preserve = int(math.ceil(num_preserve * 1. / self.preserve_round) * self.preserve_round)
+            num_preserve = int(
+                math.ceil(num_preserve * 1. / self.preserve_round) * self.preserve_round)
             if num_preserve > num_total:
-                num_preserve = int(math.floor(num_total * 1. / self.preserve_round) * self.preserve_round)
+                num_preserve = int(math.floor(
+                    num_total * 1. / self.preserve_round) * self.preserve_round)
             num_prune = num_total - num_preserve
 
         if num_total < 2 or num_prune < 1:
             return mask
         # weight*mask_weight: apply base mask for iterative pruning
-        return self.get_mask(mask, weight*mask_weight, num_prune, wrapper, wrapper_idx)
+        return self.get_mask(mask, weight*mask_weight, num_prune, wrapper, wrapper_idx, channel_masks)
```

Comment:
what is the difference between `mask` and `channel_masks`?

Reply:
`channel_masks`  indicates the output channels that we should at least mask in this conv layer. `channel_masks` actually are the common channels that all the layers in the dependency group should prune. `mask` is just the mask of this conv layer, maybe we can change the name to make this clearer?

Reply:
yes, you can directly remove channel_masks in this call, and also remove it from `_normal_calc_mask`'s argument list